bitwarden_license/src/Commercial.Infrastructure.EntityFramework/SecretsManager/Repositories/SecretRepository.cs

```diff
@@ -139,16 +139,22 @@ public async Task<Core.SecretsManager.Entities.Secret> UpdateAsync(Core.SecretsM
                 .Include("Projects")
                 .FirstAsync(s => s.Id == secret.Id);
 
-            foreach (var p in entity.Projects?.Where(p => mappedEntity.Projects.All(mp => mp.Id != p.Id)))
+            if (entity.Projects != null)
             {
-                entity.Projects.Remove(p);
-            }
+                foreach (var p in entity.Projects.Where(p => mappedEntity.Projects.All(mp => mp.Id != p.Id)))
+                {
+                    entity.Projects.Remove(p);
+                }
 
-            // Add new relationships
-            foreach (var project in mappedEntity.Projects?.Where(p => entity.Projects.All(ep => ep.Id != p.Id)))
-            {
-                var p = dbContext.AttachToOrGet<Project>(_ => _.Id == project.Id, () => project);
-                entity.Projects.Add(p);
+                if (mappedEntity.Projects != null)
+                {
+                    // Add new relationships
+                    foreach (var project in mappedEntity.Projects.Where(p => entity.Projects.All(ep => ep.Id != p.Id)))
+                    {
+                        var p = dbContext.AttachToOrGet<Project>(_ => _.Id == project.Id, () => project);
+                        entity.Projects.Add(p);
+                    }
+                }
```

Comment:
Should be good to just remove the null-conditional operator. 
Entity framework will return an empty collection of count 0 here, not null.

Same thing with the `mappedEntity` if the request is for removing project association.

```
  foreach (var p in entity.Projects.Where(p => mappedEntity.Projects.All(mp => mp.Id != p.Id)))
  {
      entity.Projects.Remove(p);
  }

  // Add new relationships
  foreach (var project in mappedEntity.Projects.Where(p => entity.Projects.All(ep => ep.Id != p.Id)))
  {
      var p = dbContext.AttachToOrGet<Project>(_ => _.Id == project.Id, () => project);
      entity.Projects.Add(p);
  }
```